content/40.units_and_quantities.md

```diff
@@ -1,269 +1,682 @@
+>>> ds = yt.load("GasSloshing/sloshing_nomag2_hdf5_plt_cnt_0100")
+>>> sp = ds.sphere("c", (100.0, "kpc"))
+>>> print(sp["gas","velocity_x"])
+[-4793397.  -5297494.5 -4997635.  ... 10608344.  10382381.  10529207. ] cm/s
+>>> print(sp["gas","velocity_x"].in_units("mile/hr"))
+[-107225.23711525 -118501.57704009 -111793.91105941 ...  237301.89692198
+  232247.24856836  235531.65264853] mile/hr
+>>> print(sp["gas","pressure"])
+[1.62223415e-10 1.60880725e-10 1.62334618e-10 ... 1.54101079e-10
+ 1.52756530e-10 1.53220436e-10] dyn/cm**2
+>>> print(sp["gas","pressure"].to("J/m**3"))
+[1.62223415e-11 1.60880725e-11 1.62334618e-11 ... 1.54101079e-11
+ 1.52756530e-11 1.53220436e-11] J/m**3
+```
 
-We provide sympy `Symbol` objects for the base dimensions.
-The dimensionality of all other units should be `sympy` `Expr` objects made up of the base dimension objects and the `sympy` operation objects `Mul` and `Pow`.
+Conversions to other unit systems can also be achieved with the methods `in_cgs` for Gaussian CGS
+units, `in_mks` for SI units, and the more general `in_base` for conversion to any valid unit
+system (see [Unit systems](#sec:unit-systems)). 
 
-Let's use some common units as examples: gram (`g`), erg (`erg`), and solar mass per cubic megaparsec (`Msun / Mpc`$^3$).
-`g` is an atomic, CGS base unit, `erg` is an atomic unit in CGS, but is not a base unit, and `Msun/Mpc`$^3$ is a combination of atomic units, which are not in CGS, and one of them even has an SI prefix.
-The dimensions of `g` are `mass` and the cgs factor is 1.
-The dimensions of `erg` are `mass * length$^2`$ * `time`$^{-2}$ and the cgs factor is 1.
-The dimensions of `Msun/Mpc`$^3$ are `mass / length`$^3$ and the cgs factor is about 6.8e-41.
+```python
+>>> import unyt as u 
+>>> import numpy as np
+>>> l = np.array([10, 100, 100])*u.Mpc
+>>> print(l.in_cgs())
+[3.08567758e+25 3.08567758e+26 3.08567758e+26] cm
+>>> print(l.in_mks())
+[3.08567758e+23 3.08567758e+24 3.08567758e+24] m
+>>> print(l.in_base("imperial"))
+[1.01236141e+24 1.01236141e+25 1.01236141e+25] ft
+>>> print(l.in_base("geometrized"))
+```
 
-We use the `UnitRegistry` class to define all valid atomic units.
-All unit registries contain a unit symbol lookup table (dict) containing the valid units' dimensionality and cgs conversion factor.
-Here is what it would look like with the above units:
+To convert an array in-place instead of making a copy, use `convert_to_units`:
+
+```python
+>>> momentum_y = sp["gas","mass"]*sp["gas","velocity_y"]
+>>> print(momentum_y)
+[1.45031068e+47 1.40467746e+47 1.42059875e+47 ... 1.01206589e+47
+ 9.94539437e+46 1.06969711e+47] cm*g/s
+>>> momentum_y.convert_to_units("Msun*kpc/Myr")
+>>> print(momentum_y)
+[745945.76032301 722474.98923595 730663.86660289 ... 520541.05697294
+ 511526.58818102 550182.82006007] Msun*kpc/Myr
+```
+
+Naturally, attempting to convert an array to units with a different dimension raises an error (in
+most cases, see below for exceptions):
+
+```python
```

Comment:
I know this "language" is supported by GitHub markdown, not sure about the flavour that's used here
```suggestion
```python-traceback
```